dohq_artifactory/admin.py

```diff
@@ -261,20 +262,104 @@ def _read_response(self, response):
         self._realm = response.get("realm")
 
     @property
-    def encryptedPassword(self):
-        if self.password is None:
+    def api_key(self):
+        """
+        Get an API key for the current user
+        :return: (str) API key
+        """
+        response = self._authenticated_user_request(
+            api_url="/api/security/apiKey", request_type=self._session.get
+        )
+        api_key = json.loads(response).get("apiKey", None)
+
+        return api_key
+
+    def api_key_create(self):
+        """
+        Create an API key for the current user.
+        Returns an error if API key already exists - use api_key_regenerate to regenerate API key instead.
+        :return: (str) API key
+        """
+        response = self._authenticated_user_request(
+            api_url="/api/security/apiKey", request_type=self._session.post
+        )
+
+        api_key = json.loads(response)["apiKey"]
+
+        return api_key
+
+    def api_key_regenerate(self):
+        """
+        Regenerate an API key for the current user
+        :return: (str) API key
+        """
+
+        if self.api_key is None:
             raise ArtifactoryException(
-                "Please, set [self.password] before query encrypted password"
+                "API key does not exist for {}. Please use api_key_create".format(
+                    self.name
+                )
             )
-        logging.debug("User get encrypted password [{x.name}]".format(x=self))
-        request_url = self._artifactory.drive + "/api/security/encryptedPassword"
-        r = self._session.get(
+
+        response = self._authenticated_user_request(
+            api_url="/api/security/apiKey", request_type=self._session.put
+        )
+        api_key = json.loads(response).get("apiKey", None)
+
+        return api_key
+
+    def api_key_revoke(self, delete_all=False):
```

Comment:
`delete_all` looks like you'll delete all API keys for a single user, but it removes every tokens, right? 
In this case - let's create a new method for it and make it obvious - `delete_api_keys_for_all_users`

Reply:
Agree, to avoid mistakes better to be more explicit, will push new changes